### PR TITLE
Fix operation name to be handler name

### PR DESCRIPTION
**Pull request by elbuo8**

### CODE REVIEW

File: lib/tracer_middleware.js [hunk 1]
```diff
@@ -49,7 +49,10 @@ function hapiHelper(tracer) {
     const response = req.response;
     const statusCode = response.isBoom ? response.output.statusCode : response.statusCode;
 
+    // set operation name before closing primary span to used parse route name
+    req.a0trace.span.setOperationName(req.route.path);
```
Comment (dafortune) [inline lib/tracer_middleware.js]: Will req.route be always available including for aborted or invalid requests?

File: lib/tracer_middleware.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
 const onFinished = require('on-finished');
-const url = require('url');
-
 
 // createRequestSpan creates a new Span for an incoming request.
 function createRequestSpan(tracer, req) {
```
Comment (dafortune) [inline lib/tracer_middleware.js]: Should we send the actual URL apart from the route?

File: lib/tracer_middleware.js [hunk 3]
```diff
@@ -49,7 +49,10 @@ function hapiHelper(tracer) {
     const response = req.response;
     const statusCode = response.isBoom ? response.output.statusCode : response.statusCode;
 
+    // set operation name before closing primary span to used parse route name
+    req.a0trace.span.setOperationName(req.route.path);
     // set final request status on the primary span.
+    req.a0trace.span.setTag(h.tracer.Tags.HTTP_URL, req.path);
```
Comment (dm03514) [inline lib/tracer_middleware.js]: What do you think about updating this to:

`req.a0trace.span.setTag(`http.path`, req.path);`

I think `http.url` (`HTTP_URL`) has special meaning in open tracing:

https://github.com/opentracing/specification/blob/master/semantic_conventions.md


-- | -- | --
http.url | string | URL of the request being handled in this segment of the trace, in standard URI format. E.g., "https://domain.net/path/to?resource=here"
-- | -- | --

I"m 99.9% sure that it wont' affect anything as is right now though

File: test/tracer.test.js [hunk 4]
```diff
@@ -189,12 +189,14 @@ describe('tracer express middleware', function() {
...
-        .get('/moreinfo')
+        .get('/moreinfo/1')
         .expect(200)
         .expect(function() {
           const report = $mock.report();
           assert.equal(1, report.spans.length);
-          assert.ok(report.firstSpanWithTagValue('moreinfo', 'here'));
+          const reqSpan = report.firstSpanWithTagValue('moreinfo', 'here');
+          assert.ok(reqSpan);
+          assert.equal('/moreinfo/:id', reqSpan.operationName());
```
Comment (dm03514) [inline test/tracer.test.js]: <3

**Comment by dm03514** [comment]
LGTM!

**Pull request merged**